Assets/Plugins/HEBGraph/Scripts/HEBGraph.cs

```diff
@@ -0,0 +1,192 @@
+{
+	public class HEBGraph : SerializedMonoBehaviour
+	{
+		const float GROUP_BG_SPACING = 0.1f;
+
+		[SerializeField] TextAsset m_HEBConfigFile;
+		[SerializeField] GameObject m_entryPrefab;
+		[SerializeField] Transform m_entryParent;
+		[SerializeField] GameObject m_groupTextPrefab, m_groupTextPrefabDown;
+		[SerializeField] Transform m_groupTextParent;
+		[SerializeField] GameObject m_linePrefab;
+		[SerializeField] GameObject m_modalPrefab;
+		[SerializeField] UIRadialDrawer m_groupBackground;
+		[SerializeField] Transform m_lineParent;
+		[SerializeField] int m_lineSections;
+		[SerializeField] ColourAsset m_defaultLineColour;
+		[SerializeField] Dictionary<int, ColourAsset> m_severityColors;
+		[SerializeField] Dictionary<int, ColourAsset> m_severityHLColors;
+
+		[HideInInspector] public Action<string> m_linkClickCallback;
+		Dictionary<int, HEBGraphLeaf> m_leaves;
+		Dictionary<(int lowId, int highId), UILineDrawer> m_lines;
+		List<HEBGraphBranch> m_branches;
+		float m_totalRadialOffset;
+		HEBGraphLeaf m_selectedLeaf;
+		BezierDrawer m_bezierDrawer;
+		//ModalManager m_modal;
+
+		//public ModalManager Modal => m_modal;
+
+		void Start()
+		{
+			m_bezierDrawer = new BezierDrawer();
+			HEBGraphData data = JsonConvert.DeserializeObject<HEBGraphData>(m_HEBConfigFile.text);
+
+			m_leaves = new Dictionary<int, HEBGraphLeaf>();
+			m_branches = new List<HEBGraphBranch>();
+			m_lines = new Dictionary<(int low, int high), UILineDrawer>();
+			List<(float start, float end)> gaps = new List<(float start, float end)>();
+			List<float> gapStarts = new List<float>();
+
+			//Create groups and items
+			foreach (HEBGraphDataGroup group in data.groups)
+			{
+				m_branches.Add(new HEBGraphBranch(group, this)); //This adds radial offsets for all created leaves
+				gapStarts.Add(m_totalRadialOffset);
+				m_totalRadialOffset += 2f;
+			}
+
+			//Position elements and gaps
+			foreach(HEBGraphBranch branch in m_branches)
+			{
+				branch.SetPosition(m_totalRadialOffset, this);
+			}
+			for (int i = 0; i < gapStarts.Count; i++)
+			{
+				gaps.Add(((gapStarts[i] + 0.5f) / m_totalRadialOffset * Mathf.PI * 2f, (gapStarts[i] + 1.5f) / m_totalRadialOffset * Mathf.PI * 2f));
+			}
+			m_groupBackground.SetGaps(gaps);
+
+			//Assign links and create lines
+			foreach (HEBGraphDataLink link in data.links)
+			{
+				if (link.toId < link.fromId ? !m_lines.ContainsKey((link.toId, link.fromId)) : !m_lines.ContainsKey((link.fromId, link.toId)))
+				{
+					UILineDrawer line = GameObject.Instantiate(m_linePrefab, m_lineParent).GetComponent<UILineDrawer>();
+					SetLineToLink(line, link);
+					m_lines.Add(link.toId < link.fromId ? (link.toId, link.fromId) : (link.fromId, link.toId), line);
+				}
+				m_leaves[link.fromId].AddLink(link);
+				m_leaves[link.toId].AddLink(link);
+			}
+
+			//Create modal manager
+			//Canvas canvas = GetComponentInParent<Canvas>();
+			//m_modal = Instantiate(m_modalPrefab, canvas.transform).GetComponent<ModalManager>();
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)